### DISCUSSION TITLE
How to plot engineering stress-strain curve?

### DISCUSSION THREAD

**Discussion by pshen20127** [idaholab/moose#26291] [Q&A Modules: Solid mechanics]
Hi All

Do you know how to plot engineering stress-strain curve of the entire model?

Thanks
Pengfei

**Comment by GiudGiud** [comment]
I think @hugary1995 answered this last time but I cannot find the post unfortunately

**Reply by hugary1995** [reply]
@pshen20127,  "entire model" is a little vague. Are you looking at a macroscale structure or a representative volume element?

**Reply by pshen20127** [reply]
I am looking at macroscale structure

**Reply by hugary1995** [reply]
The stresses and strains are different at each point in the structure, so you need to properly define what you meant by "stress strain curve for the entire model".

**Reply by pshen20127** [reply]
I am trying to plot global stress-strain curve. As the figure shown below, the stress is the average of reaction forces at the nodes on the right or left edge divided by the length of that edge. And the strain is the displacement of the right edge divided by the original length of the model.
![图片4](https://github.com/idaholab/moose/assets/81499506/4bbe04aa-bab5-4ce6-bb7a-c8d20e13c3db)

**Reply by hugary1995** [reply]
Use a NodalSum PP to get the total reaction force on the left boundary. Use a ParsedPlstprocessor to divide the reaction force by the left side length. Use a AverageNodalVariableValue PP to get the average displacement on the right side. Use a ParsedPlstprocessor to divide the average displacement by the axial length.

**Reply by pshen20127** [reply]
Got it. Do you know how to extract reaction force at a certain node?
